Online schema change best practices (#16942)
cockroachdb/docs: v22.2/online-schema-changes.md

## Best practices for online schema changes

### Estimate your storage capacity before performing online schema changes

> Some schema change operations like adding or dropping columns, or altering primary keys will result in temporary increases to a cluster's storage consumption. These operations will result in consuming up to three times the storage space of the range size while the schema change is being applied, which may cause the cluster to run out of storage space and pause or fail to apply the schema change.

Reviewer: ```suggestion
Some schema change operations, like adding or dropping columns or altering primary keys, will temporarily increase a cluster's storage consumption. Specifically, these operations may temporarily require up to three times more storage space  for the range size while the schema change is being applied, and this may cause the cluster to run out of storage space or fail to apply the schema change.
```

> Some schema change operations like adding or dropping columns, or altering primary keys will result in temporary increases to a cluster's storage consumption. These operations will result in consuming at least three times the storage space of the range size while the schema change is being applied, which may cause the cluster to run out of storage space and pause or fail to apply the schema change.

Reviewer 2: At least one's times, and at most 3 times

Author: Done.

                    (crdb_internal.index_span((SELECT id FROM system.namespace WHERE name = 'vehicles'), x.row_number)::STRING[])[2]
                )::BYTES
          AND end_key
              > (
                    (crdb_internal.index_span((SELECT id FROM system.namespace WHERE name = 'vehicles'), x.row_number)::STRING[])[1]
                )::BYTES;
    ~~~

    ~~~
                  index_name               | range_size_mb
    ----------------------------------------+----------------
      vehicles_pkey                         |        0.0003
      vehicles_pkey                         |        0.0001
      vehicles_pkey                         |        0.0004
      vehicles_pkey                         |        0.0006
      vehicles_pkey                         |        0.0002
      vehicles_pkey                         |        0.0001
      vehicles_pkey                         |        0.0001
      vehicles_pkey                         |        0.0001
      vehicles_pkey                         |        0.0014
      vehicles_auto_index_fk_city_ref_users |        0.0014
    (10 rows)
    ~~~

### Run schema changes with large backfills during off-peak hours

> Online schema changes that result in large backfill operations (for example, [`ALTER TABLE ... ALTER COLUMN`](alter-table.html#alter-column) statements) are computationally expensive, and can result in degraded performance. The [admission control system](admission-control.html) will help keep high-priority operations running, but it's better to run backfill-heavy schema changes during times with low loads.

Reviewer: ```suggestion
Online schema changes that result in large backfill operations (for example, [`ALTER TABLE ... ALTER COLUMN`](alter-table.html#alter-column) statements) are computationally expensive, and can result in degraded performance. The [admission control system](admission-control.html) will help keep high-priority operations running, but it's recommended to run backfill-heavy schema changes during times when the cluster is under relatively low loads.
```
